### PR TITLE
MAV_CMD_CONDITION_YAW - shortest direction v2

### PR DESCRIPTION
**Pull request by hamishwillee**
@rmackay9 @AndKe This PR replaces #1962 with how I think this should look based on our discussions. I am pretty sure it matches what ArduPilot does currently.

Are you OK with this?

### CODE REVIEW

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -1360,9 +1360,9 @@
       </entry>
       <entry value="115" name="MAV_CMD_CONDITION_YAW" hasLocation="false" isDestination="false">
         <description>Reach a certain target angle.</description>
-        <param index="1" label="Angle" units="deg">target angle, 0 is north</param>
-        <param index="2" label="Angular Speed" units="deg/s">angular speed</param>
-        <param index="3" label="Direction" minValue="-1" maxValue="1" increment="2">direction: -1: counter clockwise, 1: clockwise</param>
+        <param index="1" label="Angle" units="deg" minValue="-360" maxValue="360">target angle. Absolute angles: [0-360], 0 is north. Relative angles [-360 to +360], 0 current yaw.</param>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml:1363]:
```suggestion
        <param index="1" label="Angle" units="deg" minValue="-360" maxValue="360">target angle. Absolute angles: [0-360], 0 is north. Relative angles [-360 to +360], 0 is initial yaw.</param>
```

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -1360,9 +1360,9 @@
       </entry>
       <entry value="115" name="MAV_CMD_CONDITION_YAW" hasLocation="false" isDestination="false">
         <description>Reach a certain target angle.</description>
-        <param index="1" label="Angle" units="deg">target angle, 0 is north</param>
-        <param index="2" label="Angular Speed" units="deg/s">angular speed</param>
-        <param index="3" label="Direction" minValue="-1" maxValue="1" increment="2">direction: -1: counter clockwise, 1: clockwise</param>
+        <param index="1" label="Angle" units="deg" minValue="-360" maxValue="360">target angle. Absolute angles: [0-360], 0 is north. Relative angles [-360 to +360], 0 is initial yaw.</param>
```
Comment (rmackay9) [inline message_definitions/v1.0/common.xml:1363]: Hi @hamishwillee, thanks for this.  I think the minValue should remain as 0 shouldn't it?  At the moment AP doesn't actually support negative angles for the relative angles.  We could change it so it does that though.

Comment (hamishwillee) [reply message_definitions/v1.0/common.xml:1363]: Yes. I'm a dumbass. Changed to:
```suggestion
        <param index="1" label="Angle" units="deg" minValue="0" maxValue="360">target angle [0-360]. Absolute angles: 0 is north. Relative angle: 0 is initial yaw. Direction set by param3.</param>
```

**Review by rmackay9 — APPROVED** [review]
Looks good, thanks!

**Comment by hamishwillee** [comment]
@julianoes @auturgy As discussed in dev call, are you OK with this (Randy is)? It matches the ArduPilot behaviour/expectation and the common expectation from several years ago.

If so I am happy to merge - please approve.

**Review by auturgy — APPROVED** [review]

**Pull request merged**